src/Sylius/Bundle/PayumBundle/Extension/UpdatePaymentStateExtension.php

```diff
@@ -55,8 +55,16 @@ public function onExecute(Context $context)
      */
     public function onPostExecute(Context $context)
     {
-        if ($context->getPrevious()) {
+        $previousStack = $context->getPrevious();
+        $stackSize = count($previousStack);
+        
+        if ($stackSize > 1) {
             return;
+        } elseif ($stackSize === 1) {
```

Comment:
Coding style: we can replace this `elseif` with a regular `if`, these conditions are not coupled to each other.